protocol/superc20/mirror-factory-limited.md

```diff
@@ -0,0 +1,90 @@
+# Summary
+
+*The work present below is part of the [Interoperability project](https://github.com/ethereum-optimism/optimism/issues/10899).*
+
+A factory to spin up `ISuperchainERC20` Mirrors of `OptimismMintableERC20Tokens` is required. This document aims to provide a design for such a factory. The Mirrors need to be proxied to address any future addition of features. The factory will deploy these proxies. Therefore, from now on, when we say `Mirror` we refer to the Proxy and not the implementation.
```

Comment:
I would really appreciate an overall architecture/glossary section in one of these design docs, there are a lot of new terms and I want to make sure the overall design is unambiguous:
- OptimismMintableERC20Factory is a proxy, with a corresponding implementation
- OptimismMintableERC20Factory deploys OptimismMintableERC20 contracts, and the OptimismMintableERC20s are not proxied
- MirrorERC20Factory is a proxy, with a corresponding implementation
- It deploys MirrorERC20 tokens, which are proxied using the beacon pattern, so we can upgrade all implementations if needed

(Something like that, I don't know if that's exactly correct)

Reply:
Perfect, agreed. Will add it.